torch/fx/experimental/symbolic_shapes.py

```diff
@@ -4240,6 +4254,19 @@ def evaluate_guards_for_args(self, placeholders, args, *, ignore_static=True):
             return self.evaluate_guards_expression(code, args)
         return True
 
+    def get_input_guards(self, symints):
```

Comment:
Nit: don't call this `get_input_guards`. Maybe something like `get_pruned_guards`, doc'ed as "Get a list of guards, but pruned so it only provides guards that reference symints from the passed in input." TBH, it would also be more logical for this to take a set of symbols instead of a list of torch.SymInt. In any case, a type annotation helps.